### PR TITLE
Feat(page): Add exporting components and APIs for page component to support page template extension (alpha version)

### PR DESCRIPTION
**Pull request by betterdancing**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced `PageGeneral` for improved page structure.
	- Added dynamic component rendering in `PageSetting` for enhanced flexibility.
	- Implemented confirmation dialogs for unsaved changes during page copying.

- **Improvements**
	- Enhanced page creation process with asynchronous behavior for better user experience.
	- Introduced a new reactive reference for managing selected template cards.
	- Improved validation checks before saving page settings.

### CODE REVIEW

File: packages/plugins/page/index.js [hunk 1]
```diff
@@ -13,11 +13,17 @@
...
 import { PageService } from './src/composable'
+import PageGeneral from './src/PageGeneral.vue'
+import PageMain from './src/PageMain.vue'
 
 export default {
   ...metaData,
   apis: api,
   entry,
+  components: {
+    PageMain,
```
Comment (hexqi) [inline packages/plugins/page/index.js]: PageMain不需要定义到components里，entry本身就是PageMain了，如果需要定制用户可以直接替换entry

Comment (betterdancing) [reply packages/plugins/page/index.js]: PageMain已经删掉，还原为之前的Main.vue了

**Review by hexqi — DISMISSED** [review]

**Review by chilingling — DISMISSED** [review]

File: packages/plugins/page/index.js [hunk 2]
```diff
@@ -13,11 +13,16 @@
 import entry, { api } from './src/Main.vue'
 import metaData from './meta.js'
 import { PageService } from './src/composable'
+import PageGeneral from './src/PageGeneral.vue'
+import { api as pageSettingApi } from './src/PageSetting.vue'
 
 export default {
   ...metaData,
-  apis: api,
+  apis: [...api, ...pageSettingApi],
```
Comment (hexqi) [inline packages/plugins/page/index.js]: apis是对象格式，不是数组

**Review by hexqi — COMMENTED** [review]
PR的标题尽量用英文

**Review by hexqi — APPROVED** [review]

**Pull request merged**